pkg/scalers/nats_jetstream_scaler.go

```diff
@@ -0,0 +1,262 @@
+	DeliveryStatus consumerDeliveryStatus `json:"delivery"`
+}
+
+type consumerConfig struct {
+	DurableName string `json:"durable_name"`
+}
+
+type consumerDeliveryStatus struct {
+	ConsumerSequence int64 `json:"customer_seq"`
+	StreamSequence   int64 `json:"stream_seq"`
+}
+
+func NewNATSJetStreamScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	jsMetadata, err := parseNATSJetStreamMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing NATS JetStream metadata: %s", err)
+	}
+
+	return &natsJetStreamScaler{
+		metricType: metricType,
+		stream:     &streamDetail{},
+		metadata:   jsMetadata,
+		httpClient: kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false),
+	}, nil
+}
+
+func parseNATSJetStreamMetadata(config *ScalerConfig) (natsJetStreamMetadata, error) {
+	meta := natsJetStreamMetadata{}
+	var err error
+	meta.monitoringEndpoint, err = GetFromAuthOrMeta(config, "natsServerMonitoringEndpoint")
+	if err != nil {
+		return meta, err
+	}
+
+	if config.TriggerMetadata["account"] == "" {
+		return meta, errors.New("no account name given")
+	}
+	meta.account = config.TriggerMetadata["account"]
+
+	if config.TriggerMetadata["stream"] == "" {
+		return meta, errors.New("no stream name given")
+	}
+	meta.stream = config.TriggerMetadata["stream"]
+
+	if config.TriggerMetadata["consumer"] == "" {
+		return meta, errors.New("no consumer name given")
+	}
+	meta.consumer = config.TriggerMetadata["consumer"]
+
+	meta.lagThreshold = defaultJetStreamLagThreshold
+
+	if val, ok := config.TriggerMetadata[lagThresholdMetricName]; ok {
+		t, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return meta, fmt.Errorf("error parsing %s: %s", lagThresholdMetricName, err)
+		}
+		meta.lagThreshold = t
+	}
+
+	meta.activationLagThreshold = 0
+	if val, ok := config.TriggerMetadata["activationLagThreshold"]; ok {
+		activationTargetQueryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return meta, fmt.Errorf("activationLagThreshold parsing error %s", err.Error())
+		}
+		meta.activationLagThreshold = activationTargetQueryValue
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return meta, nil
+}
+
+func (s *natsJetStreamScaler) getNATSJetStreamEndpoint() string {
+	return fmt.Sprintf("http://%s/jsz?acc=%s&consumers=true&config=true", s.metadata.monitoringEndpoint, s.metadata.account)
+}
+
+func (s *natsJetStreamScaler) IsActive(ctx context.Context) (bool, error) {
+	monitoringEndpoint := s.getNATSJetStreamEndpoint()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, monitoringEndpoint, nil)
+	if err != nil {
+		return false, err
+	}
+
+	resp, err := s.httpClient.Do(req)
+	if err != nil {
+		jsLog.Error(err, "Unable to access NATS JetStream monitoring endpoint", "natsServerMonitoringEndpoint", s.metadata.monitoringEndpoint)
```

Comment:
```suggestion
		jsLog.Error(err, "unable to access NATS JetStream monitoring endpoint", "natsServerMonitoringEndpoint", s.metadata.monitoringEndpoint)
```

Reply:
please use lowercase at the beginning of error messages, the same apply for the other error messages in this scaler.